streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -170,8 +170,10 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
     ssc.waiter.notifyError(e)
   }
 
-  private class JobHandler(job: Job) extends Runnable {
+  private class JobHandler(job: Job) extends Runnable with Logging {
     def run() {
+      ssc.sc.setLocalProperty(JobScheduler.BATCH_TIME_PROPERTY_KEY, job.time.milliseconds.toString)
+      ssc.sc.setLocalProperty(JobScheduler.OUTPUT_OP_ID_PROPERTY_KEY, job.outputOpId.toString)
```

Comment:
This setting and clearing of property needs to be done in a `try....finally` so that it gets cleared in case of any exception.